app/javascript/components/filters/Filters.vue

```diff
@@ -49,18 +54,26 @@
 
         // create an empty array for each filter
         this.filters.forEach(filter => {
+
           if (filter.name !== undefined && filter.options.length > 0) {
+            const preselected = this.presetOptionsForFilter(filter.name)
+
             let obj = {}
 
             obj.name = filter.name
-            obj.options = []
+            obj.options = preselected
             obj.type = filter.type
 
             array.push(obj)
           }
         })
 
         this.$store.dispatch('table/setFilterOptions', array)
+      },
+
+      presetOptionsForFilter(filterName) {
+        const preselected = this.presetFilters.filter(presetFilter => presetFilter.name === filterName)
+        return preselected.length > 0 ? preselected[0]['options'] : []
       }
```

Comment:
https://github.com/unepwcmc/action-agenda/pull/79#discussion_r840366316 - ah! In this case it looks like you're only interested in the first result (`preselected[0]`), so you can use `find` instead of `filter` so that it stops if it finds a result

```
        const preselected = this.presetFilters.find(presetFilter => presetFilter.name === filterName)
        return preselected ? preselected.options : []
```

If the compiler is using ES2020 or later, you should be able to simplify the second line to something like `return preselected?.options || []` using [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining), but I have a feeling this repo won't be that up to date. You could try it and see if it breaks...

Reply:
it did not break! changed